benchmark/storage/index_wrapper_benchmark.cpp

```diff
@@ -0,0 +1,154 @@
+  storage::RecordBufferSegmentPool buffer_pool_{1000000, 1000000};
+  catalog::Schema table_schema_;
+  catalog::IndexSchema index_schema_;
+
+ public:
+  std::default_random_engine generator_;
+
+  const uint32_t table_size_ = 10000000;
+
+  // SqlTable
+  storage::SqlTable *sql_table_;
+  storage::ProjectedRowInitializer tuple_initializer_ =
+      storage::ProjectedRowInitializer::Create(std::vector<uint8_t>{1}, std::vector<uint16_t>{1});  // This is a dummy
+
+  // HashIndex
+  common::ManagedPointer<storage::index::Index> index_;
+  transaction::TimestampManager *timestamp_manager_;
+  transaction::DeferredActionManager *deferred_action_manager_;
+  transaction::TransactionManager *txn_manager_;
+
+  byte *key_buffer_;
+
+ protected:
+  void SetUp(const benchmark::State &state) override {
+    auto col = catalog::Schema::Column(
+        "attribute", type::TypeId::INTEGER, false,
+        parser::ConstantValueExpression(type::TransientValueFactory::GetNull(type::TypeId::INTEGER)));
+    StorageTestUtil::ForceOid(&(col), catalog::col_oid_t(1));
+    table_schema_ = catalog::Schema({col});
+    sql_table_ = new storage::SqlTable(&block_store_, table_schema_);
+    tuple_initializer_ = sql_table_->InitializerForProjectedRow({catalog::col_oid_t(1)});
+
+    timestamp_manager_ = new transaction::TimestampManager;
+    deferred_action_manager_ = new transaction::DeferredActionManager(timestamp_manager_);
+    txn_manager_ = new transaction::TransactionManager(timestamp_manager_, deferred_action_manager_, &buffer_pool_,
+                                                       true, DISABLED);
+    gc_ = new storage::GarbageCollector(timestamp_manager_, deferred_action_manager_, txn_manager_, DISABLED);
+    gc_thread_ = new storage::GarbageCollectorThread(gc_, gc_period_);
+  }
+  void TearDown(const benchmark::State &state) override {
+    gc_thread_->GetGarbageCollector().UnregisterIndexForGC(index_);
+
+    delete gc_thread_;
+    delete gc_;
+    delete sql_table_;
+    delete[] key_buffer_;
+    delete txn_manager_;
+    delete deferred_action_manager_;
+    delete timestamp_manager_;
+  }
+
+  void CreateIndex(const storage::index::IndexType type) {
+    std::vector<catalog::IndexSchema::Column> keycols;
+    keycols.emplace_back("", type::TypeId::INTEGER, false,
+                         parser::ColumnValueExpression(CatalogTestUtil::TEST_DB_OID, CatalogTestUtil::TEST_TABLE_OID,
+                                                       catalog::col_oid_t(1)));
+    StorageTestUtil::ForceOid(&(keycols[0]), catalog::indexkeycol_oid_t(1));
+    index_schema_ = catalog::IndexSchema(keycols, type, false, false, false, true);
+
+    index_ = (storage::index::IndexBuilder().SetKeySchema(index_schema_)).Build();
+    gc_thread_->GetGarbageCollector().RegisterIndexForGC(index_);
+    key_buffer_ = common::AllocationUtil::AllocateAligned(index_->GetProjectedRowInitializer().ProjectedRowSize());
+  }
+
+  void PopulateTableAndIndex() {
```

Comment:
Table is populated as sequentially increasing keys. On the hash index this shouldn't make a performance difference, but on the BwTree it might.